File: backend/models/course.go
package models

import "gorm.io/gorm"

// Модель Course
type Course struct {
	gorm.Model
	Title       string `gorm:"size:255;not null;" json:"title"`
	Category    string `gorm:"size:255;not null;" json:"category"`
	Description string `gorm:"size:255;not null;" json:"description"`
	Video       string `gorm:"size:255;not null;" json:"video"`
	UserId      uint   `gorm:"not null;" json:"user_id"`
	TestId      *uint  `json:"test_id"` // Nullable поле
	Test        *Test  `gorm:"foreignKey:TestId;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;" json:"test"`
}

File: frontend-2/lib/features/feed/domain/course.dart
class Course {
  final int id;
  final String title;
  final DateTime createdAt;
  final String? category;
  final String? description;
  final int? testId;

  Course({
    required this.id,
    required this.title,
    required this.createdAt,
    this.category,
    this.description,
    this.testId,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['ID'],
      title: json['title'],
      createdAt: DateTime.parse(json['CreatedAt']),
      category: json['category'],
      description: json['description'],
      testId: json['test_id'],
    );
  }

  bool get hasTest => testId != null;
}